### PR TITLE
Fix issue #25 

### LINKED ISSUE #25: Final plots with miniTrees
We need a simple script making plots running analysis.cpp
Plots should be put in a directory with the git commit well identified.
Plots are meant for final AN and also as validation of each production. It's a crucial instrument to make sure that whenever a new version of the code is deployed, it produces consistent results.
The script should also run on the miniTrees and dump in a txt file the number of events for each tree. This a very quick check on the sanity of the miniTree production and validation of the code.

### PR DESCRIPTION
**Pull request by skalafut**
fixes #25 and another issue to count the events in the Z mass window when processing dytagandprobe minitrees.

### CODE REVIEW

File: bin/analysis.cpp [hunk 1]
```diff
@@ -443,7 +444,7 @@ int main(int ac, char* av[])
 					}
 
 					Fits::massWR.setVal(selEvent.WR_mass);
-					Fits::evtWeight.setVal(selEvent.weight);
+					Fits::genEvtWeights.setVal(selEvent.weight);
```
Comment (shervin86) [inline bin/analysis.cpp]: we have discussed about that, this has to be removed

File: bin/analysis.cpp [hunk 2]
```diff
@@ -458,12 +459,9 @@ int main(int ac, char* av[])
 			if(i == seed) std::cout << zMass80to100EvtCount << "\tevents from the dataset named\t" << selEvent.datasetName << "\tpass isPassingLooseCuts and have 81.2 < dilepton_mass < 101.2" << std::endl;
 			if(i == seed) std::cout << zMass85to95EvtCount << "\tevents from the dataset named\t" << selEvent.datasetName << "\tpass isPassingLooseCuts and have 86.2 < dilepton_mass < 96.2" << std::endl;
 
-			std::endl;
-
-			std::endl;
 
 			///make a permanent RooDataSet which has the same information as tempDataSet, but with events which are weighted according to the var named evtWeight
-			RooDataSet * permanentWeightedDataSet = new RooDataSet("permanentWeightedDataSet", "permanentWeightedDataSet", tempDataSet, Fits::vars, "", Fits::evtWeight.GetName());
+			RooDataSet * permanentWeightedDataSet = new RooDataSet("permanentWeightedDataSet", "permanentWeightedDataSet", tempDataSet, Fits::vars, "", Fits::genEvtWeights.GetName());
```
Comment (shervin86) [inline bin/analysis.cpp]: same here

File: configs/miniTreeEntries.dat [hunk 3]
```diff
@@ -0,0 +1,26 @@
+priorCommitNumberForWhichTheseEventCountNumbersAreValid
+e6114abb9badb083f04b5c9433e721cb1c078b8d
+[?1034h# Dataset miniTree_flavoursideband miniTree_lowdileptonsideband miniTree_signal_ee miniTree_signal_mumu miniTree_dytagandprobe
```
Comment (shervin86) [inline configs/miniTreeEntries.dat]: If you manage to understand where this bad thing arises, would be great. The commit part should be preceded by a # at the beginning of the line, such that can be discarded by other scripts

File: scripts/wrValidation.sh [hunk 4]
```diff
@@ -0,0 +1,76 @@
+#!/bin/bash
+
+###cd to cmsWR/. and run the python script test/miniTreeDumpCheck.py to check the number of entries in minitrees
+eval "echo priorCommitNumberForWhichTheseEventCountNumbersAreValid >& configs/miniTreeEntries.dat"
```
Comment (shervin86) [inline scripts/wrValidation.sh]: why "eval"? Only stdout should be redirected to file in my opinion

**Comment by shervin86** [comment]
There are too many plotter macros, it's difficult to track what is used and why. Can you put some comments at the beginning of the macros that are used in order to explain what they do? I would like to understand if there are things that are simple duplicates and that can be merged into a more reasonable macro/program with options or not.

**Comment by skalafut** [comment]
I added comments to the plotting macros and wrValidation.sh to explain why there are two plotting macros, made other updates to wrValidation.sh, committed the changes, and called git rebase -i master in branch tempIssueTwentyFive.

**Comment by shervin86** [comment]
Can you add in a separate PR the documentation with the instructions?

**Comment by skalafut** [comment]
yes I will submit a separate pull request with instructions tonight.


Regards,
Sean Kalafut

Physics PhD candidate
University of Minnesota Twin Cities

**Pull request merged**